src/sentry/sentry_metrics/multiprocess.py

```diff
@@ -536,22 +538,92 @@ def __init__(
         self.__producer_topic = settings.KAFKA_TOPICS[settings.KAFKA_SNUBA_METRICS].get(
             "topic", "snuba-metrics"
         )
+        self.__commit_function = commit_function
 
         self.__closed = False
         self.__metrics = get_metrics()
+        self.__produced_message_offsets = {}
+        self.__callbacks = 0
+        self.__started = time.time()
+        # Need to make these flags
+        self.__commit_max_batch_size = 10000
+        self.__commit_max_batch_time = 4.0
+        self.__producer_queue_max_size = 80000
+        self.__producer_long_poll_timeout = 3.0
+
+        # poll duration metrics
+        self.__poll_start_time = time.time()
+        self.__poll_duration_sum = 0.0
+        self.__poll_sample_batch = 0
+
+    def _ready(self) -> bool:
+        if self.__callbacks >= self.__commit_max_batch_size:
+            return True
+        now = time.time()
+        if now >= (self.__started + self.__commit_max_batch_time):
+            return True
+
+        return False
+
+    def _record_poll_duration(self, poll_duration) -> None:
+        self.__poll_duration_sum += poll_duration
+
+        # record poll time durations every 5 seconds
+        if (self.__poll_start_time + 5) < time.time():
+            self.__metrics.incr(
+                "simple_produce_step.join_duration", amount=int(self.__poll_duration_sum)
+            )
+            self.__poll_duration_sum = 0
+            self.__poll_start_time = time.time()
+
+    def poll_producer(self, timeout: float) -> None:
+        # XXX(meredith): There are two metrics we are trying to capture:
+        #  1. How long are we spending per poll. Let's only record this metric for every 100 messages.
+        if self.__poll_sample_batch >= 100:
+            with self.__metrics.timer("produce_step.producer_poll_duration"):
+                self.__producer.poll(timeout)
+            self.__poll_sample_batch = 0
+            return
+        #  2. On avg whats the time spent polling in total. This excludes
+        #     the time for polls that we capture above ^, I didn't want the
+        #     datadog metrics included in the start and end times. This metric
+        #     will use "simple_produce_step.join_duration" name so that we can
+        #     compare easily in datadog.
+        start = time.time()
+        self.__producer.poll(timeout)
+        end = time.time()
+        poll_duration = end - start
+        self._record_poll_duration(poll_duration)
```

Comment:
This is simpler 
the metrics object has
`def timer(key, instance=None, tags=None, sample_rate=settings.SENTRY_METRICS_SAMPLE_RATE):`

Just call this method every time and pass a sample rate you want.
That ssampling rate will be passed down to datadog which will multiply the result by the sampling rate when displaying them thus leaving everything consistent.

Reply:
okay so basically instead just have one metric like the following? (and forget about the "produce_step.producer_poll_duration")

```python
with self.__metrics.timer("simple_produce_step.join_duration", sample_rate=0.05):
    self.__producer.poll(timeout)
```